src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -530,9 +528,13 @@ private VariantContext addNonRefSymbolicAllele(final VariantContext mergedVC) {
         final List<VariantContext> eventsAtThisLoc = new ArrayList<>();
 
         if( activeAllelesToGenotype.isEmpty() ) {
-            haplotypes.stream().map(h -> h.getEventMap().get(loc))
-                    .filter(vc -> vc != null && eventsAtThisLoc.stream().noneMatch(vc::hasSameAllelesAs))
-                    .forEach(eventsAtThisLoc::add);
+            final List<VariantContext> events = haplotypes.stream().map(h -> h.getEventMap().get(loc))
+                    .filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
Spent another 20 minutes and couldn't come up with an alternative solution. Moving on.